src/LoginOAuth2.test.js

```diff
@@ -1,68 +1,92 @@
-      redirectUri="http://foo.test/auth/github"
-      authorizeUri="http://bar.test"
-    />
-  );
-  let tree = component.toJSON();
-
-  expect(tree).toMatchSnapshot();
-});
-
-test('Renders with `className`', () => {
-  const component = renderer.create(
-    <LoginOAuth2
-      clientId="foo"
-      redirectUri="http://foo.test/auth/github"
-      authorizeUri="http://bar.test"
-      className="foobar"
-    />
-  );
-  let tree = component.toJSON();
-
-  expect(tree).toMatchSnapshot();
-});
-
-test('Renders with `buttonText`', () => {
-  const component = renderer.create(
-    <LoginOAuth2 clientId="foo"
-      authorizeUri="http://bar.test"
-      redirectUri="http://foo.test/auth/github"
-      buttonText="Foo"
-    />
-  );
-  let tree = component.toJSON();
+import { fireEvent, render } from '@testing-library/react';
+import { findInstance } from 'react-dom-instance';
 
-  expect(tree).toMatchSnapshot();
-});
-
-test('Opens OAuth dialog', () => {
-  const clientId = 'foo';
-  const redirectUri = 'http://foo.test/auth/github';
-
-  const component = (
-    <LoginOAuth2
-      clientId={clientId}
-      redirectUri={redirectUri}
-      state="hello"
-      authorizeUri="http://bar.test"
-    />
-  );
-  const wrapper = shallow(component);
-
-  wrapper.find('button').simulate('click');
-
-  const query = `client_id=${clientId}&scope=email&redirect_uri=${redirectUri}&response_type=code&state=hello`
+import LoginOAuth2 from './LoginOAuth2.js';
 
-  expect(wrapper.instance().popup.url).toBe(
-    `http://bar.test?${query}`
-  );
-});
+describe('LoginOauth2', () => {
+  it('should render defaults', () => {
+    const { container, getByRole } = render(
+      <LoginOAuth2
+        clientId="foo"
+        redirectUri="http://foo.test/auth/github"
+        authorizeUri="http://bar.test"
+      />
+    );
+    
+    expect(container.firstChild).toMatchSnapshot();
+  
+    const button = getByRole('button', {
+      name: 'Sign in',
+      hidden: false,
+    });
+  
+    expect(button).toBeEnabled;
+    expect(button.getAttribute('class')).toBe('');
+  });
+  
+  it('should render with `className`', () => {
+    const { container, getByRole } = render(
+      <LoginOAuth2
+        clientId="foo"
+        redirectUri="http://foo.test/auth/github"
+        authorizeUri="http://bar.test"
+        className="foobar"
+      />
+    );
+  
+    const button = getByRole('button', {
+      name: 'Sign in',
+      hidden: false,
+    });
+  
+    expect(button.getAttribute('class')).toBe('foobar');
+  });
+  
+  it('should render with `buttonText`', () => {
+    const buttonText = 'Foo';
+  
+    const { container, getByRole } = render(
+      <LoginOAuth2 clientId="foo"
+        authorizeUri="http://bar.test"
+        redirectUri="http://foo.test/auth/github"
+        buttonText={buttonText}
+      />
+    );
+  
+    const button = getByRole('button', {
+      name: buttonText,
+      hidden: false,
+    });
+  
+    expect(button).toBeEnabled;
```

Comment:
```suggestion
    expect(button).toBeEnabled();
```

Reply:
Adopted your changes, but tests get now [a type error](https://github.com/okteto/react-oauth2-login/actions/runs/4045460658/jobs/6956961153).

Reply:
I thought `toBeEnabled` was a function, but actually it doesn't even exist, it is `undefined` 😅

I always use `not.toBeDisabled()` instead since it is the more common way, it should work.